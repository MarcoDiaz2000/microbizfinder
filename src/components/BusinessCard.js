import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const BusinessCard = ({ business }) => (
  <div
    className="business-card"
    data-testid="business-card"
    style={{ backgroundImage: `url(${business.image_url})` }}
  >
    <h2>{business.name}</h2>
    <p>{business.categories.map((category) => category.title).join(', ')}</p>
    <p>{`Rating: ${business.rating} / 5`}</p>
    <Link to={`/details/${business.id}`}>See details</Link>
  </div>
);

BusinessCard.propTypes = {
  business: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    image_url: PropTypes.string,
    categories: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string,
    })),
    location: PropTypes.shape({
      address1: PropTypes.string,
      city: PropTypes.string,
    }),
    rating: PropTypes.number,
  }).isRequired,
};

export default BusinessCard;
