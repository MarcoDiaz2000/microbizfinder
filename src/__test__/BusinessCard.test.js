import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import BusinessCard from '../components/BusinessCard';

test('BusinessCard matches snapshot', () => {
  const business = {
    id: '1',
    name: 'Example Business',
    image_url: 'example.com/image.jpg',
    categories: [{ title: 'Category 1' }, { title: 'Category 2' }],
    location: { address1: '123 Main St', city: 'Example City' },
    rating: 4.5,
  };

  const { container } = render(
    <MemoryRouter>
      <BusinessCard business={business} />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});
