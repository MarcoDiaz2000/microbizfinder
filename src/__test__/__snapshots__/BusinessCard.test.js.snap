// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessCard matches snapshot 1`] = `
<div>
  <div
    class="business-card"
    data-testid="business-card"
    style="background-image: url(example.com/image.jpg);"
  >
    <h2>
      Example Business
    </h2>
    <p>
      Category 1, Category 2
    </p>
    <p>
      Rating: 4.5 / 5
    </p>
    <a
      href="/details/1"
    >
      See details
    </a>
  </div>
</div>
`;
